src/ProgressOnderwijsUtils/DistinctArray.cs

```diff
@@ -8,44 +8,98 @@ namespace ProgressOnderwijsUtils
 {
     public static class DistinctArray
     {
-        public static DistinctArray<T> FromDistinct<T>([NotNull] IEnumerable<T> items)
-            => DistinctArray<T>.FromDistinct(items);
+        [Pure]
+        public static DistinctArray<T> ToDistinctArray<T>([NotNull] this IEnumerable<T> items)
+            => DistinctArray<T>.FromPossiblyNotDistinct(items, EqualityComparer<T>.Default);
+
+        [Pure]
+        public static DistinctArray<T> ToDistinctArray<T>([NotNull] this ISet<T> items)
+            => ToDistinctArrayFromDistinct_Unchecked(items.ToArray());
+
+        [Pure]
+        public static DistinctArray<T> ToDistinctArray<T, TVal>([NotNull] this Dictionary<T, TVal>.KeyCollection items)
+            => ToDistinctArrayFromDistinct_Unchecked(items.ToArray());
+
+        [Pure]
+        public static DistinctArray<T> ToDistinctArray<T>([NotNull] this IEnumerable<T> items, IEqualityComparer<T> comparer)
+            => items is HashSet<T> set && (object)set.Comparer == (object)comparer ? set.ToDistinctArray() : DistinctArray<T>.FromPossiblyNotDistinct(items, comparer);

```

Comment:
Waarom niet ```ISet<T>```?

Reply:
ISet heeft geen Comparer (en het is toch maar een optimalisatie).